moshi/src/main/java/com/squareup/moshi/Types.kt

```diff
@@ -212,11 +212,17 @@ public object Types {
       is Class<*> -> {
         return if (b is GenericArrayType) {
           equals(a.componentType, b.genericComponentType)
+        } else if (b is ParameterizedType && a.rawType == b.rawType) {
+          // Class instance with generic info, from method return types
+          return a.typeParameters.map { it.bounds }.toTypedArray().flatten() == b.actualTypeArguments.toList()
```

Comment:
Can we use this?

```suggestion
          return a.typeParameters.flatMap { it.bounds } == b.actualTypeArguments.toList()
```

Reply:
Yes, but I had to change it.bounds to it.bounds.toList(), because flatMap only takes Iterable, not Array.